src/formatDate.js

```diff
@@ -9,6 +9,46 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    dateMap[fromFormat[i]] = dateParts[i];
+  }
+
+  const year =
+    dateMap['YYYY'] ||
+    (dateMap['YY']
+      ? (parseInt(dateMap['YY'], 10) < 30 ? '20' : '19') + dateMap['YY']
+      : null);
```

Comment:
You can simplify the ternary operation by using template literals for the year conversion. This would make the code easier to read and understand.